src/main/java/org/jitsi/videobridge/AbstractEndpoint.java

```diff
@@ -325,6 +325,46 @@ public JSONObject getDebugState()
         return debugState;
     }
 
+    /**
+     * Computes and sets the {@link #maxReceiverVideoConstraints} from the
+     * specified video constraints.
+     *
+     * @param newVideoConstraints the map of receiver endpoint id -> video
+     * constraints that specifies who (which receiver endpoint) is viewing what
+     * (as determined by the video constraints) from this endpoint (which is the
+     * sender).
+     */
+    private void setReceiverVideoConstraints(
+        ImmutableMap<String, VideoConstraints> newVideoConstraints)
+    {
+        VideoConstraints oldReceiverMaxVideoConstraints
+            = this.maxReceiverVideoConstraints;
+
+        VideoConstraints defaultVideoConstraints
+            = new VideoConstraints(BitrateControllerConfig.Config.thumbnailMaxHeightPx());
+
+        VideoConstraints newReceiverMaxVideoConstraints = newVideoConstraints
+            .values()
+            .stream()
+            .max(Comparator.comparingInt(VideoConstraints::getIdealHeight))
+            .orElse(defaultVideoConstraints);
```

Comment:
It looks like we only care about `idealHeight`, but the event is fired any time the `VideoConstrant` instances don't match. Maybe fire the event if `newReceivedMaxVideoConstaints.getIdealHeight() != oldReceiverMaxVideoConstraints.getIdealHeight()`?

Reply:
Yes, and I think this emphasizes how awkward the `preferredHeight` and `preferredFps` are in the VideoConstraints, which takes us back to @bbaldino suggestion to have a separate object for bandwidth allocation preferences/policy.

Reply:
@bbaldino @bgrozev I've taken the preferred fps/resolution out of the video constraints class and put it in a video allocation policy here ef9e0270.

Three motivations behind this change:

1.  It doesn't feel right to signal the preferred fps/resolution to the sender for layer suspension
2. @bbaldino has been talking about an allocation policy for a while now and I agree with that, I just didn't know where/how to fit it
3. @bgrozev's comment here really nailed it: in the future when we extend the video constraints class with an ideal frame rate we need to go come back here to fix a bug.

Let me know what you think